### PR TITLE
Created DialogHelper for managing dialogues in a mvvmFx application

### PR DESCRIPTION
**Pull request by mainrs**

### CODE REVIEW

File: mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+					stage.close();
+				}
+			});
+		}
+
+
+		return stage;
+	}
+
+	// don't really know why this if statement exists. I kept it because there might be a bug that I am not
```
Comment (manuel-mauky) [inline mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java:48]: In some cases you like to reopen an existing dialog. In this case you don't need to create a new Stage but can reuse the old one.

File: mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	// aware of. But creating a new Scene each time the method is called shouldn't be a problem.
+	private static void configureScene(Stage stage, Supplier<Parent> content, String... css) {
+		// no idea why this if statement is here. Doesn't make much sense to me. Just create a new Scene every time
+		// and we are fine
+		if (stage.getScene() == null) {
+			// create new scene and apply css
+			Scene scene = new Scene(content.get());
```
Comment (manuel-mauky) [inline mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java:55]: The idea of the `Supplier<Parent>` is that you can lazy-load the actual view. For this reason we should only call the supplier when the dialog will be shown (determined by the provided boolean property). If you look into the original DialogHelper in the examples you can find this behaviour

File: mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Created by Sven on 21/06/16.
+ */
+public class DialogHelper {
+
+	// prevent instantiation from outside
+	private DialogHelper() {}
+
+	public static Stage initDialog(Window owner, Supplier<Parent> content, BooleanProperty visibleProperty, String...
```
Comment (manuel-mauky) [inline mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java:20]: I think we should find a better name for "visibleProperty". In the original DialogHelper the name "openProperty" was used which isn't really a better name. 
May "showProperty"? It's what actually is happening (stage.show())

Comment (mainrs) [inline mvvmfx/src/main/java/de/saxsys/mvvmfx/utils/dialogs/DialogHelper.java:20]: I agree with it. Thanks for the explanations above 😄

**Comment by jpitchardu** [comment]
Any update on this?

**Comment by manuel-mauky** [comment]
Hi Jonathan, 
I have no updates on this topic. We had a long discussion in #331 on this but haven't gotten any further then a prototype yet. This topic had a low priority for us and therefor I wasn't working on it for some time. 

Do you need a special dialog handling? If so, for a first workaround you could copy+paste the `DialogsHelper` class from this pull request or the contacts-example in the repository (link: https://github.com/sialcasa/mvvmFX/blob/develop/examples/contacts-example/src/main/java/de/saxsys/mvvmfx/examples/contacts/util/DialogHelper.java) and try it out. Any feedback is welcome.

**Comment by jpitchardu** [comment]
Yeah, I did that thanks.

**Pull request closed without merging**